src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -75,8 +75,11 @@ public ProjectRestoreCommand(ILogger logger, ProjectRestoreRequest request)
                 _request.MaxDegreeOfConcurrency,
                 token);
 
+            // The user global package folder is always ordered first.
+            var userPackageFolder = localRepositories.First();
```

Comment:
This is a little bit of magic IMO, could we make a type LocalRepositories with a UserPackageFolder prop?